spec/models/vm_scan_spec.rb

```diff
@@ -1,4 +1,67 @@
 RSpec.describe VmScan do
+  context "A single VM Scan Job," do
+    let(:server) { EvmSpecHelper.local_miq_server(:has_vix_disk_lib => true) }
+    let(:user) { FactoryBot.create(:user_with_group, :userid => "tester") }
+    let(:ems) { FactoryBot.create(:ems_redhat_with_authentication, :zone => server.zone) }
+    let(:vm) { FactoryBot.create(:vm_redhat, :ext_management_system => ems, :host => host, :miq_group => user.current_group, :evm_owner => user) }
+    let(:host) { FactoryBot.create(:host, :ext_management_system => ems) }
+    let(:job) { described_class.first }
+
+    describe "#scan" do
+      before do
+        allow(MiqEventDefinition).to receive_messages(:find_by => true)
+        allow(server).to receive(:has_active_role?).with('automate').and_return(true)
+
+        vm.scan
+
+        job_item = MiqQueue.find_by(:class_name => "MiqAeEngine", :method_name => "deliver")
+        job_item.delivered(*job_item.deliver)
+      end
+
+      it "should start in a state of waiting_to_start" do
+        expect(job.state).to eq("waiting_to_start")
+      end
+
+      describe "#start" do
+        it "should check policy" do
+          job.signal(:start)
+          expect(job.reload.state).to eq("checking_policy")
+        end
+
+        it "should transit through scanning, synchronizing, and finished states" do
```

Comment:
Note: I want to split this test up into individual state transitions, this just let me test the whole run through

Reply:
since your test is focused on RHV, should it live in that provider repo?  We can move it after this PR is merged, but I am just asking for clarity.

Reply:
Actually I was hoping for this to not be RHV focused and just test the core state machine.  I picked RHV because they don't override any of the states like VMware does but I get also get the tests to work with the generic VmInfra and EmsInfra factories so I'll switch over to those so it doesn't look RHV specific.

Reply:
Okay I split this up into individual tests per state transition